data/helpers.d/utils

```diff
@@ -487,3 +485,69 @@ ynh_check_app_version_changed () {
   fi
   echo $return_value
 }
+
+# Compare the current package version against another version given as an argument.
+# This is really useful when we need to do some actions only for some old package versions.
+#
+# example: ynh_compare_current_package_version --comparison gt --version 2.3.2~ynh1
+# This example will check if the installed version is greater than (gt) the version 2.3.2~ynh1
+#
+# Generally you might probably use it as follow in the upgrade script
+#
+# if ynh_compare_current_package_version --comparaison gt --version 2.3.2~ynh1
+# then
+#     # Do something that is needed for the package version older than 2.3.2~ynh1
+# fi
+#
+# usage: ynh_compare_current_package_version --comparison lt|gt|le|ge
+# | arg: --comparison - Comparison type. Could be : le (lower than), gt (greater than), le (lower or equal), ge (greater or equal)
+# | arg: --version - The version to compare. Need to be a version in the yunohost package version type (like 2.3.1~ynh4)
+#
+# Return 0 if the evaluation is true. 1 if false.
+#
+# Requires YunoHost version 3.8.0 or higher.
+ynh_compare_current_package_version() {
+    local legacy_args=cv
+    declare -Ar args_array=( [c]=comparison= [v]=version= )
+    local version
+    local comparison
+    # Manage arguments with getopts
+    ynh_handle_getopts_args "$@"
+
+    local current_version=$YNH_APP_CURRENT_VERSION
+
+    # Check the syntax of the versions
+    if [[ ! $version =~ '~ynh' ]] || [[ ! $current_version =~ '~ynh' ]]
+    then
+        ynh_print_warn "Invalid agument for version."
+        return 1
+    fi
+
+    # If the version are identical, and the evaluation allows equal versions.
+    if [ $version == $current_version ]
+    then
+        if [ $comparison == ge ] || [ $comparison == le ]; then
+            return 0
+        else
+            return 1
+        fi
+    fi
+
+    # Check if the current version is greater than the one given as argument
+    if [ $comparison == ge ] || [ $comparison == gt ]
+    then
+        if [ $(printf "$version\n$current_version" | sort --version-sort | tail -n 1) == $current_version ]; then
+            return 0
+        else
+            return 1
+        fi
+
+    # Else if the current version is lower than the one given as argument
+    else
+        if [ $(printf "$version\n$current_version" | sort --version-sort | tail -n 1) == $version ]; then
```

Comment:
I'm randomly jumping in but wouldn't it be simpler to just use `dpkg --compare-versions` ?

Reply:
Indeed http://man7.org/linux/man-pages/man1/dpkg.1.html
Probably more secure

Reply:
Oh yes. Sow this but I thought that wasn't useable as a simple tools. Now it should be better